src/components/Main/Main.js

```diff
@@ -0,0 +1,61 @@
+import axios from 'axios';
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Outlet, Link } from 'react-router-dom';
+import { fetchCoinsFromAPI } from '../../redux/main';
+import Coin from './Coin';
+import crypto from '../../assets/images/cryptocurrency.png';
+
+function Main() {
+  const coins = useSelector((state) => state.main);
+  const dispatch = useDispatch();
+
+  const [coinCount, setCoinCount] = useState('');
+  const [activeMarkets, setActiveMarkets] = useState('');
+  const [totalVolume, setTotalVolume] = useState('');
+
+  const displayGlobalCoins = () => {
+    axios.get('https://api.coinlore.net/api/global/')
+      .then((response) => {
+        setCoinCount(`${response.data[0].coins_count}`);
+        setActiveMarkets(`${response.data[0].active_markets}`);
+        setTotalVolume(`$${response.data[0].total_volume}`);
+      });
+  };
+
+  useEffect(() => {
+    if (coins.length === 0) { dispatch(fetchCoinsFromAPI()); }
+    displayGlobalCoins();
+  }, []);
+
+  const content = coins.map((coin) => (<Link to={`/details/${coin.id}`} key={coin.id}><Coin coin={coin} /></Link>));
```

Comment:
- [ ] One of the project requirements is to implement a filter stateless component to filter data retrieved from the API, this should be done using a controlled component(search input). The list of items shown should be the result of filtering through the coins.